lib/compare.js

```diff
@@ -24,24 +24,23 @@ const equalObject = (obj1, obj2) => {
     const key = keys1[i];
     const val1 = obj1[key];
     const val2 = obj2[key];
-    if (val1 == val2) continue; // eslint-disable-line eqeqeq
```

Comment:
Do we need this check? This seems to be the same as the first check inside the `equal` function.